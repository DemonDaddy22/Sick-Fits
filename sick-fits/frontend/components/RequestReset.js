import React from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';

export const REQUEST_RESET_MUTATION = gql`
    mutation REQUEST_RESET_MUTATION($email: String!) {
        requestReset(email: $email) {
            message
        }
    } 
`;

export default class RequestReset extends React.Component {

    state = {
        email: ''
    }

    handleChange = e => this.setState({ [e.target.name]: e.target.value });

    render = () => <Mutation mutation={REQUEST_RESET_MUTATION} variables={this.state}>
        {(reset, { error, loading, called }) => <Form data-test='form' method='POST' onSubmit={async e => {
            e.preventDefault();
            await reset();
            this.setState({ email: '' });
        }}>
            <fieldset disabled={loading} aria-busy={loading}>
                <h2>Request to reset your password</h2>
                <Error error={error} />
                {!error && !loading && called && <p>Success! Check your email for a reset link!</p>}
                <label htmlFor='email'>Email</label>
                <input id='email' type='email' name='email' placeholder='Your email' value={this.state.email} onChange={this.handleChange} />
                <button type='submit'>Request Reset</button>
            </fieldset>
        </Form>}
    </Mutation>;
}