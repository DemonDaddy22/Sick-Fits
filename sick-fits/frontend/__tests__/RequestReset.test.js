import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';
import { MockedProvider } from 'react-apollo/test-utils';

const mocks = [
    {
        request: { query: REQUEST_RESET_MUTATION, variables: { email: 'foo@bar.com' } },
        result: { data: { requestReset: { message: 'success', __typename: 'Message' } } }
    }
];

describe('<RequestReset />', () => {
    it('renders and matches snapshot', () => {
        const wrapper = mount(<MockedProvider mocks={mocks}>
            <RequestReset />
        </MockedProvider>);

        const form = wrapper.find('[data-test="form"]');
        // console.log(form.debug());
        expect(toJSON(form)).toMatchSnapshot();
    });

    it('calls the mutation', async () => {
        const wrapper = mount(<MockedProvider mocks={mocks}>
            <RequestReset />
        </MockedProvider>);

        wrapper.find('input').simulate('change', { target: { name: 'email', value: 'foo@bar.com' } });
        wrapper.find('form').simulate('submit');
        await wait();
        wrapper.update();
        expect(wrapper.find('p').text()).toContain('Success!');
    });
});