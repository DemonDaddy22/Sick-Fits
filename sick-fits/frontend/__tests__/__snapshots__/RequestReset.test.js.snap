// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> renders and matches snapshot 1`] = `
Array [
  <Form
    data-test="form"
    method="POST"
    onSubmit={[Function]}
  >
    <form
      className="Form-sc-1xszr8q-0 dUkfCz"
      data-test="form"
      method="POST"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Request to reset your password
        </h2>
        <DisplayError
          error={Object {}}
        />
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Your email"
          type="email"
          value=""
        />
        <button
          type="submit"
        >
          Request Reset
        </button>
      </fieldset>
    </form>
  </Form>,
  <form
    className="Form-sc-1xszr8q-0 dUkfCz"
    data-test="form"
    method="POST"
    onSubmit={[Function]}
  >
    <fieldset
      aria-busy={false}
      disabled={false}
    >
      <h2>
        Request to reset your password
      </h2>
      <DisplayError
        error={Object {}}
      />
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Your email"
        type="email"
        value=""
      />
      <button
        type="submit"
      >
        Request Reset
      </button>
    </fieldset>
  </form>,
]
`;
